tfhe/src/high_level_api/integers/types/compact.rs

```diff
@@ -90,3 +92,284 @@ where
         })
     }
 }
+
+impl<P: IntegerParameter> ParameterSetConformant for GenericCompactInteger<P> {
+    type ParameterSet = PBSParameters;
+    fn is_conformant(&self, pbs_parameters: &PBSParameters) -> bool {
+        let params = CtListConformanceParams {
+            shortint_params: pbs_parameters.to_shortint_conformance_param(),
+            num_blocks_per_integer: P::num_blocks(),
+            num_integers_constraint: ListSizeConstraint {
+                max_group_count: 1,
+                group_size: 1,
+            },
+        };
+        self.list.is_conformant(&params)
+    }
+}
+
+impl<P: IntegerParameter> Named for GenericCompactInteger<P> {
+    const NAME: &'static str = "high_level_api::GenericCompactInteger";
+}
+
+impl<P: IntegerParameter> ParameterSetConformant for GenericCompactIntegerList<P> {
+    type ParameterSet = (PBSParameters, ListSizeConstraint);
+    fn is_conformant(
+        &self,
+        (pbs_parameters, number_integers_constraint): &(PBSParameters, ListSizeConstraint),
+    ) -> bool {
+        let params = CtListConformanceParams {
+            shortint_params: pbs_parameters.to_shortint_conformance_param(),
+            num_blocks_per_integer: P::num_blocks(),
+            num_integers_constraint: *number_integers_constraint,
+        };
+        self.list.is_conformant(&params)
+    }
+}
+
+#[cfg(test)]
```

Comment:
missing the enabled feature for safe deser ?

I think we can enable it by default if shortint or integer are enabled

I guess we could put

#[cfg(any(feature = shortint, feature = integer))] for the safe deserialization module ? this way it's available and bincode is enabled at the same time ?

or keep the current safe_serialization feature but shortint and integer enable that feature automatically

Reply:
this may be still relevant if we don't auto enable the safe deserialization with shortint/integer automatically

Reply:
`conformance` is not part of `safe-deserialization` so it's not feature gated

Reply:
ah right, thanks